conv.c

```diff
@@ -28,30 +28,20 @@ numeric_to_symbolic(uint16_t num) {
 
 uint16_t
 symbolic_to_numeric(char *str) {
-	char special[] = "stST";
-
 	uint16_t numeric = 0;
-	for (unsigned int field = 0; field < 3; field++) {
-		for (unsigned int bit = 0; bit < 3; bit++) {
-			if (str[field * 3 + bit] != '-') {
-				if (specialp) {
-					for (unsigned int i = 0; i < LEN(special); i++) {
-						if (str[field * 3 + bit] == special[i]) {
-							numeric += 01000 * (1 << (2 - field));
-							break;
-						}
-					}
-					if (str[field * 3 + bit] != 'S' &&
-							str[field * 3 + bit] != 'T') {
-						numeric += expt(8, 2 - field) * (1 << (2 - bit));
-					}
-				}
-				else numeric += expt(8, 2 - field) * (1 << (2 - bit));
+
+	for (unsigned bit = 0; bit < 9; bit++) {
```

Comment:
I'm not really a fan of ommiting the `int` type. Without it, at first sight, it seems like `unsigned bit` is a type.